test/functional/ui/highlight_spec.lua

```diff
@@ -16,6 +17,91 @@ describe('color scheme compatibility', function()
   end)
 end)
 
+describe('buffer switching leaves manual syntax highlights ok', function()
+  -- When using manual syntax highlighting, it should be preserved even when
+  -- switching buffers... bug did only occur without :set hidden
+  -- Ref: vim patch 7.4.1236
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new()
+    screen:attach()
+    --ignore highligting of ~-lines
+    screen:set_default_attr_ignore( {{bold=true, foreground=Screen.colors.Blue}} )
+    --syntax highlight for vimcscripts "echo"
+    screen:set_default_attr_ids( {[1] = {bold=true, foreground=Screen.colors.Brown}} )
+  end)
+
+  after_each(function()
+    screen:detach()
+    os.remove('tmp2.vim')
```

Comment:
Test artifacts should be named like `Xtest-{test name}-...`. E.g. here it could be named `Xtest-functional-ui-highlight.tmp.vim`.